src/core/server/saved_objects/import/create_saved_objects.ts

```diff
@@ -76,7 +81,70 @@ export const createSavedObjects = async <T>({
   );
 
   // filter out the 'version' field of each object, if it exists
+
   const objectsToCreate = filteredObjects.map(({ version, ...object }) => {
+    if (dataSourceId) {
+      // @ts-expect-error
+      if (dataSourceTitle && object.attributes.title) {
+        if (
+          object.type === 'dashboard' ||
+          object.type === 'visualization' ||
+          object.type === 'search'
+        ) {
+          // @ts-expect-error
+          object.attributes.title = object.attributes.title + `_${dataSourceTitle}`;
+        }
+      }
+
+      if (object.type === 'index-pattern') {
+        object.references = [
+          {
+            id: `${dataSourceId}`,
+            type: 'data-source',
+            name: 'dataSource',
+          },
+        ];
+      }
+
+      if (object.type === 'visualization' || object.type === 'search') {
+        // @ts-expect-error
+        const searchSourceString = object.attributes?.kibanaSavedObjectMeta?.searchSourceJSON;
+        // @ts-expect-error
+        const visStateString = object.attributes?.visState;
+
+        if (searchSourceString) {
+          const searchSource = JSON.parse(searchSourceString);
```

Comment:
what if `searchSourceString` is not valid json?

Reply:
Hi, Bandini, thanks for the suggestions, but the file uploading already handle the invalid case, I do not think we need to add one more try catch for the validated results, [code](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/saved_objects/routes/import.ts#L87) for outer try catch

Reply:
Will this only get called from Browser? What about curl request directly?